client/features/media/MediaDetail.jsx

```diff
@@ -1,43 +1,166 @@
 import React from 'react';
 import StarRatingInteractive from '../sharedComponents/StarRatingInteractive';
+import StarRating from '../sharedComponents/StarRating';
+import SearchBar from './../sharedComponents/SearchBar.jsx';
+import Reviews from './../reviews/Reviews.jsx';
+import test from './testData.js';
 import './MediaDetail.css';
 
-const MediaDetail = props => (
-  <div id="MovieDetail">
+//figure out how to wrap with error boundary and logger
+//need to find a way to get banner images to each movie/tv show
+
+class MediaDetail extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      id: '',
+      mediaType: '',
+      name: '',
+      rating: '',
+      summary: '',
+      subscriptions: [],
+      reviews: [],
+      imgUrl: '',
+      userSubs: [],
+      watchWithSubscribed: [],
+      watchWithUnsubscribed: []
+    };
+    this.getMediaAndUserDetails = this.getMediaAndUserDetails.bind(this);
+    this.handleLogoClick = this.handleLogoClick.bind(this);
+  }
+
+  componentDidMount() {
+    this.getMediaAndUserDetails();
+  }
+
+  getMediaAndUserDetails() {
+    //this will be a request to the db to get media info to apply to state
+    let subs = test.subscriptions;
+
+    //on success, another request to db to find out what subscriptions the user has OR this can be passed to me in props
+    //I will need the userId passed to me in props
+    let userSubs = test.userSubs;
+
+    //on success do some logic
+    let services = [];
+    let nonservices = [];
+    for (let i = 0; i < subs.length; i++) {
+      if (userSubs.includes(subs[i])) {
+        services.push(subs[i]);
+      } else {
+        nonservices.push(subs[i]);
+      }
+    }
+
+    this.setState({
+      id: test.id,
+      mediaType: test.mediaType,
+      name: test.name,
+      rating: test.rating,
+      summary: test.summary,
+      subscriptions: subs,
+      reviews: test.reviews,
+      imgUrl: test.thumbnail,
+      userSubs: userSubs,
+      watchWithSubscribed: services,
+      watchWithUnsubscribed: nonservices
+    });
+  }
+
+  handleLogoClick() {
+    //notify home component that user has watched the movie?
+  }
+
+  render() {
+    let subscribed = this.state.watchWithSubscribed.map((name, i) => {
```

Comment:
nice work here - just food for thought: with subscribed and the other variable below it - you could extract the functionality into a helper file, then import the function into this file.   Not necessary and this is working, but it could help conserve on line space =)